src/sockperf.cpp

```diff
@@ -1725,9 +1725,27 @@ static int resolve_sockaddr(const char *host, const char *port, int sock_type,
     // any protocol
     hints.ai_protocol = 0;
     hints.ai_socktype = sock_type;
-
+    int res;
     struct addrinfo *result;
-    int res = getaddrinfo(host, port, &hints, &result);
+
+    if (host != NULL) {
+        std::string path = host;
+        if (path.size() > 0 && path[0] == '/') {
+            log_dbg("provided path for Unix Domain Socket is %s\n", host);
+            if (path.length() >= MAX_UDS_NAME) {
+                log_err("length of name is larger than %d bytes", MAX_UDS_NAME);
+                res = SOCKPERF_ERR_SOCKET;
+                return res;
+            }
+            addr_len = MAX_UDS_NAME;
```

Comment:
UNIX socket address contains not only a path but also a `sun_family` member. So `addr_len` must be set to `sizeof(sockaddr_store_t)`.

Reply:
When changing `addr_len = MAX_UDS_NAME` to `addr_len = sizeof(sockaddr_store_t);`
I get an issue: 
` ERROR: [fd=3] Can`t bind socket, IP to bind: /tmp/test [UNIX]`

Reply:
It's probably because you're using `addr_len` in `strncpy` call one line below. Third argument for `strncpy` must be really `MAX_UDS_NAME`. But `addrlen` is an output parameter that must be set accordingly to socket address usage rules (to `sizeof(sockaddr_store_t)`).

Reply:
ok but if I change it to `sizeof(sockaddr_store_t)` so in bringup it goes to 
`tmp->server_addr_len  = _user_params.addr_len;`

then in `server.cpp` it continues with `socklen_t bind_addr_len = g_fds_array[ifd]->server_addr_len;`
Finally when binding: 
`if (bind(ifd, reinterpret_cast<const sockaddr *>(p_bind_addr), bind_addr_len) < 0)`

if `bind_addr_len` is not set to 108 (`MAX_UDS_NAME`) it won't bind.

Reply:
Are `sizeof(struct sockaddr_un)` and `sizeof(struct sockaddr_storage_t)` different in this case?

Reply:
Yes, this is definitely an alignment issue. Try to set `addr_len = sizeof(strruct sockaddr_un)`